File: src_fortran/sub/maxii.f
c
c
c
c
c
c
      subroutine MAXII(n,nx,imax,jndex)
c determine maximum-value of an integer-array
      implicit none
      integer jndex,i,n,imax, nx(n)
      jndex=1
      do 1 i=1,n
   1  if(nx(jndex).le.nx(i)) jndex=i
      imax=nx(jndex)
      return
      end ! of subr. maxni

File: src/sub/maxii.c
int maxii(int n, const int *nx, int *imax, int *jndex) {
    int i;

    if (n <= 0 || nx == 0 || imax == 0 || jndex == 0) {
        return -1;
    }

    *jndex = 0;
    for (i = 0; i < n; ++i) {
        if (nx[*jndex] <= nx[i]) {
            *jndex = i;
        }
    }

    *imax = nx[*jndex];
    return 0;
}
